src/mcsat/solver.c

```diff
@@ -2703,6 +2748,9 @@ void mcsat_solve(mcsat_solver_t* mcsat, const param_t *params, model_t* mdl, uin
       break;
     }
 
+    // remove all the hints
+    int_queue_reset(&mcsat->hinted_decision_vars);
```

Comment:
I think it is better to move this call in the mcsat_pop_internal method. That way we will also reset the queue when we do a restart.